### PR TITLE
contrib: removal of dependency on Invenio-Groups

### LINKED ISSUE #18: contrib: remove strong dependency on Invenio-Groups
- [ ] Create a signal when the OAuth response has arrived in CERN and ORCID.
- [ ] Send the response to that signal.

### PR DESCRIPTION
**Pull request by JavierDelgadoFernandez**
* Removes the dependency on Invenio-Groups reducing the number of fields
  parsed inside CERN or ORCID modules and sending the response to a
  subscribable signal. (closes #18)

Signed-off-by: Javier Delgado <javier.delgado.fernandez@cern.ch>

### CODE REVIEW

File: invenio_oauthclient/contrib/cern.py [hunk 1]
```diff
@@ -210,13 +216,5 @@ def account_setup(remote, token):
...
-        current_user.modified = True
-        current_user.save()
-
-        if user and not any([user.family_name, user.given_names]):
-            user.family_name = res['Lastname'][0]
-            user.given_names = res['Firstname'][0]
-
-            db.session.add(user)
-            current_user.reload()
+        res['groups'] = fetch_groups(res['Group'])
```
Comment (JavierDelgadoFernandez) [inline invenio_oauthclient/contrib/cern.py]: Is this really needed? Why is necessary to parse the groups and remove some of them?

**Comment by jirikuncar** [comment]
IMHO we can create a general signal in https://github.com/inveniosoftware/invenio-oauthclient/blob/master/invenio_oauthclient/handlers.py#L259-L266 with `remote.name` and `resp`.

File: invenio_oauthclient/contrib/cern.py (hunk 1 above)
Comment (jirikuncar) [inline invenio_oauthclient/contrib/cern.py]: Because some of them should not be managed by Invenio.

File: invenio_oauthclient/contrib/cern.py [hunk 2]
```diff
@@ -159,6 +160,9 @@
 REMOTE_APP_RESOURCE_SCHEMA = "http://schemas.xmlsoap.org/claims/"
 
 
+oauth_cern_response_received = signal('oauth-cern-response-received')
```
Comment (lnielsen) [inline invenio_oauthclient/contrib/cern.py]: the signal  should be defined in a ``signals.py`` file.

Also it's better to define a generic signal ``oauth-reposnse-received`` and use the remote app as sender.

Comment (JavierDelgadoFernandez) [inline invenio_oauthclient/contrib/cern.py]: @jirikuncar From ```handlers.py``` there is no access to get all the resources of the user. Should the methods ```account_info``` and ```account_setup``` return a dictionary with all the fields?

**Comment by jirikuncar** [comment]
Please cherry-pick from https://github.com/jirikuncar/invenio-oauthclient/tree/signals

**Comment by JavierDelgadoFernandez** [comment]
@jirikuncar Ping.

File: invenio_oauthclient/handlers.py [hunk 3]
```diff
@@ -337,7 +344,10 @@ def signup_handler(remote, *args, **kwargs):
             raise OAuthError("Could not create token for user.", remote)
 
         if not token.remote_account.extra_data:
-            handlers['setup'](token, response)
+            account_setup = handlers['setup'](token, response)
+            account_setup_received.send(
+                remote, response=response, account_setup=account_setup
```
Comment (jirikuncar) [inline invenio_oauthclient/handlers.py]: you might want to pass also `token`

File: invenio_oauthclient/contrib/cern.py (hunk 2 above)
Comment (jirikuncar) [inline invenio_oauthclient/contrib/cern.py]: please remove this signal

**Comment by JavierDelgadoFernandez** [comment]
@jirikuncar Done.

File: invenio_oauthclient/contrib/orcid.py [hunk 4]
```diff
@@ -119,9 +120,13 @@
 ))
 
 
+oauth_orcid_response_received = signal('oauth-orcid-response-received')
```
Comment (jirikuncar) [inline invenio_oauthclient/contrib/orcid.py]: remove this signal

**Comment by JavierDelgadoFernandez** [comment]
@jirikuncar Done.

File: invenio_oauthclient/contrib/cern.py (hunk 2 above)
Comment (jirikuncar) [inline invenio_oauthclient/contrib/cern.py]: @JavierDelgadoFernandez please remove this signal. It is not needed anymore.

**Comment by JavierDelgadoFernandez** [comment]
@jirikuncar Done.

File: invenio_oauthclient/contrib/cern.py [hunk 5]
```diff
@@ -201,22 +199,6 @@ def account_info(remote, resp):
...
-        current_user.modified = True
-        current_user.save()
-
-        if user and not any([user.family_name, user.given_names]):
-            user.family_name = res['Lastname'][0]
-            user.given_names = res['Firstname'][0]
-
-            db.session.add(user)
-            current_user.reload()
+    return None
```
Comment (jirikuncar) [inline invenio_oauthclient/contrib/cern.py]: `pass`

**Comment by jirikuncar** [comment]
- [ ] please include `signals.py` in docs

**Comment by JavierDelgadoFernandez** [comment]
@jirikuncar Done.

**Pull request merged**